presto-orc/src/main/java/com/facebook/presto/orc/writer/LongColumnWriter.java

```diff
@@ -129,13 +134,16 @@ public void writeBlock(Block block)
         }
 
         // record values
+        int nonNullValueCount = 0;
         for (int position = 0; position < block.getPositionCount(); position++) {
             if (!block.isNull(position)) {
                 long value = type.getLong(block, position);
                 dataStream.writeLong(value);
                 statisticsBuilder.addValue(value);
+                nonNullValueCount++;
             }
         }
+        return nonNullValueCount * typeSize + (block.getPositionCount() - nonNullValueCount) * NULL_SIZE;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).